e2e/tests/plugins/imagery/exampleImagery.e2e.spec.js

```diff
@@ -0,0 +1,217 @@
+        await page.keyboard.down('Alt');
+        await page.mouse.down();
+        await page.mouse.move(imageCenterX, imageCenterY, 10);
+        await page.mouse.up();
+        await page.keyboard.up('Alt');
+        const afterLeftPanBoundingBox = await bgImageLocator.boundingBox();
+        expect(afterRightPanBoundingBox.x).toBeLessThan(afterLeftPanBoundingBox.x);
+
+        // pan up
+        await page.mouse.move(imageCenterX, imageCenterY);
+        await page.keyboard.down('Alt');
+        await page.mouse.down();
+        await page.mouse.move(imageCenterX, imageCenterY + 200, 10);
+        await page.mouse.up();
+        await page.keyboard.up('Alt');
+        const afterUpPanBoundingBox = await bgImageLocator.boundingBox();
+        expect(afterUpPanBoundingBox.y).toBeGreaterThan(afterLeftPanBoundingBox.y);
+
+        // pan down
+        await page.keyboard.down('Alt');
+        await page.mouse.down();
+        await page.mouse.move(imageCenterX, imageCenterY - 200, 10);
+        await page.mouse.up();
+        await page.keyboard.up('Alt');
+        const afterDownPanBoundingBox = await bgImageLocator.boundingBox();
+        expect(afterDownPanBoundingBox.y).toBeLessThan(afterUpPanBoundingBox.y);
+
+    });
+
+    test('Can use + - buttons to zoom on the image', async ({ page }) => {
+        const bgImageLocator = await page.locator(backgroundImageSelector);
+        await bgImageLocator.hover();
+        const zoomInBtn = await page.locator('.t-btn-zoom-in');
+        const zoomOutBtn = await page.locator('.t-btn-zoom-out');
+        const initialBoundingBox = await bgImageLocator.boundingBox();
+
+        await zoomInBtn.click();
+        await zoomInBtn.click();
+        // wait for zoom animation to finish
+        await page.waitForTimeout(300);
+        const zoomedInBoundingBox = await bgImageLocator.boundingBox();
+        expect(zoomedInBoundingBox.height).toBeGreaterThan(initialBoundingBox.height);
+        expect(zoomedInBoundingBox.width).toBeGreaterThan(initialBoundingBox.width);
+
+        await zoomOutBtn.click();
+        // wait for zoom animation to finish
+        await page.waitForTimeout(300);
+        const zoomedOutBoundingBox = await bgImageLocator.boundingBox();
+        expect(zoomedOutBoundingBox.height).toBeLessThan(zoomedInBoundingBox.height);
+        expect(zoomedOutBoundingBox.width).toBeLessThan(zoomedInBoundingBox.width);
+
+    });
+
+    test('Can use the reset button to reset the image', async ({ page }) => {
+        const bgImageLocator = await page.locator(backgroundImageSelector);
+        await bgImageLocator.hover();
+        const zoomInBtn = await page.locator('.t-btn-zoom-in');
+        const zoomResetBtn = await page.locator('.t-btn-zoom-reset');
+        const initialBoundingBox = await bgImageLocator.boundingBox();
+
+        await zoomInBtn.click();
+        await zoomInBtn.click();
+        // wait for zoom animation to finish
+        await page.waitForTimeout(300);
```

Comment:
In this case, this timing might cause flake if it's running with lower resources (i.e. in CircleCI). I might consider changing this to something like 1000 if there's no way around it. 

One of the cool thing that is built into playwright is it's ability to detect animation change https://playwright.dev/docs/actionability

In our case, we could replace the waitForTimeout with a locator 'Stable' state check. https://playwright.dev/docs/actionability#stable

There's a table at the top of the actionability API page. Not all commands against a locator will wait for stable. In our case, I think we could do:

`await bgImageLocator.hover()` _should_ do the trick. However, since this was declared as a const variable at the beginning of the test, it might have a stale reference.

If await bgImageLocatr.hover() doesn't work, I would try
`await page.locator(backgroundImageSelector).hover()`

Reply:
@unlikelyzero - Yep, that is a solid suggestio. The `.waitForTimeout` approach seemed a little hacky with magic numbers. The other alternative I considered is moving all the css animation to its own class that can be removed for testing purposes. Not sure if that would be a better alternative, though we might possibly miss any animation timing related bugs.

Reply:
@michaelrogers I'm not sure what would be involved with moving to a separate class, but could imagine that it might come with too much complexity. Did 1 or 2 happen to work?

Reply:
@unlikelyzero - Option 1 did work, I just pushed up the changes.